streams/src/test/java/org/apache/kafka/streams/kstream/RepartitionTopicNamingTest.java

```diff
@@ -496,7 +496,7 @@ private Topology buildTopology(final String optimizationConfig) {
         mappedStream.filter((k, v) -> k.equals("A"))
                 .join(countStream, (v1, v2) -> v1 + ":" + v2.toString(),
                         JoinWindows.of(Duration.ofMillis(5000L)),
-                        Joined.with(Serdes.String(), Serdes.String(), Serdes.Long(), fourthRepartitionTopicName))
+                        StreamJoined.with(Serdes.String(), Serdes.String(), Serdes.Long()).withStoreName(fourthRepartitionTopicName).withName(fourthRepartitionTopicName))
```

Comment:
As above. Compatibility?

Reply:
Same comments from above.  The test will break as we no longer name the state store from `Joined` so our choices are to change the topology description or use the new class `StreamJoined` to explicitly name the state store.